Add Secret-1.gir

czirok opened the pull request:
Hi,

I've added support for `Secret-1.gir`, along with a demo application that includes:

- Access Token  
- Microsoft Identity

Please let me know if anything needs to be improved or adjusted.

badcel commented:
Thanks for your PR and all the work you did. Its amazing! :rocket: 

Some remarks:
Adding a new library does not involve adding a copy of the gir file to this repository as those files will be added and maintained semi-automatically.

I saw that libsecret is available additionally for:
- Mac: https://formulae.brew.sh/formula/libsecret#default
- Windows: https://packages.msys2.org/base/mingw-w64-libsecret

So there is no need to add it only for linux?

As in the current GNOME [SDK 48 there is Libsecret 0.21.7](https://gitlab.gnome.org/GNOME/gnome-build-meta/-/blob/gnome-46/elements/sdk/libsecret.bst?ref_type=heads) the definition inside `gir_files.txt` should probably be:

```
Secret-1.gir,libsecret-1,0.22.0
```

This allows all future versions of the 0.21.X release to be automatically part of this repository.

Please be aware that the `gir_files.txt` file is ordered alphabetically. If you add libsecret please add it at the correct line.

Please amend your changes to your last commit so there will be only one commit in your PR.

Just as some preliminary information what I would like to happen before GirCore version 0.7.0
1. This PR gets merged (see my comments above)
2. GirFiles get updated with libsecret (semi-automatically)
3. Gir.Core will update to latest girfiles (before 0.7.0 release)
4. Gir.Core will get a new project for libsecret (PR from your side, before 0.7.0 release)
5. Gir.Core will (eventually) get a Demo app showcasing libsecret using some 0.7.0 nuget (PR from your side)

This means your current PR in the `gircore/gir.core` repo has to split up. I did not test your demo app yet. From a first glance it looks to comprehensive for me to maintain inside GirCore itself.

If you compare it to the samples your code is more like a "Demo" and not like a "Sample". The current samples involve mostly one file and are extremely simple. This is the level I can and want to maintain.

Nevertheless the code and resulting app you provided look pretty well thought through already which is great! So I would like to suggest you two ways forward:
- You could either host the demo under your own GitHub account and we add a link to the "Apps" page on the homepage or even add some new "Demo" page to the homepage. We could even add some link to the readme if you like.
- Alternatively if you see the code more actively as a part of GirCore itself I could create some new Repo like "gircore/demos" and give you write access to this repository. Then you can add your sample there and maintian it as part of GirCore organization. We would link this repo from the readme and homepage of course.

What do you think?

badcel commented:
You should be able to "rebase" your PR on the current main branch.

Let's tackle your contributions one by one.

Unfortunately there is currently no concept on how to distribute code that improves the usability of the given API.

In general I try to slim down the manually written code in each project to keep the maintenance low and be close to the C API.

Let's see how all this will work out.

czirok commented:
Guten Morgen!

I understand the concept now,  the src/Libs directory is meant to stay as close to the C API as possible, and therefore should not include usability-focused extensions.

I've removed the demo, and replaced it with a unit test project. This makes it easier to validate the functionality in the future, and can be extended if needed. Extensions like ToSchemaHashTable() or GetPassword() are now located inside the unit test project. I believe this is a good place for them, as it helps keep the testing process clean and comprehensible.

Let me know if you'd like anything adjusted or further clarified. Happy to align with the direction of the project.
